depthai_helpers/supervisor.py

```diff
@@ -0,0 +1,62 @@
+import os
+import subprocess
+import sys
+import time
+from pathlib import Path
+
+def createNewArgs(args):
+    def removeArg(name, withValue=True):
+        if name in sys.argv:
+            idx = sys.argv.index(name)
+            if withValue:
+                del sys.argv[idx + 1]
+            del sys.argv[idx]
+
+    removeArg("-gt")
+    removeArg("--guiType")
+    removeArg("--noSupervisor")
+    return sys.argv[2:] + ["--noSupervisor", "--guiType", args.guiType]
+
+
+class Supervisor:
+    def runDemo(self, args):
+        new_env = os.environ.copy()
+        if args.guiType == "qt":
+            from PyQt5.QtCore import QLibraryInfo
+            new_env["QT_QPA_PLATFORM_PLUGIN_PATH"] = QLibraryInfo.location(QLibraryInfo.PluginsPath)
+            new_env["QT_QUICK_BACKEND"] = "software"
+        new_env["DEPTHAI_INSTALL_SIGNAL_HANDLER"] = "0"
+        args.noSupervisor = True
+
+        try:
+            new_args = createNewArgs(args)
+            subprocess.check_call(sys.argv[:2] + new_args, env=new_env)
+        except subprocess.CalledProcessError as ex:
+            print("Error while running demo script... {}".format(ex))
+            print("Waiting 5s for the device to be discoverable again...")
+            time.sleep(5)
+            args.guiType = "cv"
+            new_args = createNewArgs(args)
+            subprocess.check_call(sys.argv[:2] + new_args, env=new_env)
+
+    def checkQtAvailability(self):
+        try:
+            from PyQt5.QtWidgets import QMainWindow, QVBoxLayout, QWidget, QLabel, QApplication
+            from PyQt5.QtGui import QPixmap
+            from PyQt5.QtCore import Qt, QTimer
+
+            app = QApplication([sys.argv[0]])
+
+            label = QLabel()
+            px = QPixmap(str(Path(__file__).parent / "logo.png"))
+            label.setPixmap(px)
+            label.resize(200, 200)
+            label.setWindowFlags(Qt.FramelessWindowHint)
+            label.setAttribute(Qt.WA_TranslucentBackground)
+            label.show()
+            QTimer.singleShot(10, label.close)
+            exit_code = app.exec_()
+            return exit_code == 0
```

Comment:
Does supervisor have to show anything? Can we check without that? 
Launcher will run the demo as is, will it show the logo again?

Reply:
I wanted to mimic a regular run to hopefully detect more issues that can happen. But, since it's only 10ms, the logo does not appear at all, users just experience a slight delay in demo startup

Reply:
@themarpe I removed this code eventually, as importing Qt modifies the system and can result in a segfault from Qt